ipa-core/src/secret_sharing/vector/impls.rs

```diff
@@ -84,37 +84,34 @@ macro_rules! boolean_vector {
                     let a = rng.gen::<$vec>();
                     let b = rng.gen::<$vec>();
 
-                    // required by DZKP storage strategy
-                    if (a.as_bitslice().len() % 256 == 0 || 256 % a.as_bitslice().len() == 0) {
-                        let bit_shares = bit.share_with(&mut rng);
-                        let a_shares = a.share_with(&mut rng);
-                        let b_shares = b.share_with(&mut rng);
+                    let bit_shares = bit.share_with(&mut rng);
+                    let a_shares = a.share_with(&mut rng);
+                    let b_shares = b.share_with(&mut rng);
```

Comment:
Is there a reason this test can't use the `TestWorld::semi_honest` helper, which would take care of upgrading the inputs and some of the other boilerplate? (Note that `TestWorld::semi_honest` provides a non-upgraded context, so I think the code to obtain and use a malicious validator should still work.)

Reply:
I have changed it but then decided to change it back since using TestWorld:malcious would require me to implement into shares for the struct (a,b,bit) which creates more boilerplate code. TestWorld:semi_honest does not work since the validator does not do anything for semi honest and upgraded semi honest contexts and I wanted to make sure that the Boolean multiplication adds values to the batch.

Reply:
There is an existing `IntoShares` impl for 2-tuples, so you could do `((a, b), bit)`.

> TestWorld:semi_honest does not work since the validator does not do anything for semi honest and upgraded semi honest contexts and I wanted to make sure that the Boolean multiplication adds values to the batch.

I don't understand this. What is different about the contexts that `TestWorld::semi_honest` uses than the ones you get from `world.contexts()`? It seems to me like the malicious test will actually exercise adding values to the batch, and the semi-honest test won't. And I'm not sure how the semi-honest test could add values to the batch, without effectively turning it into the malicious test.